client/payment-request/event-handlers.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Internal dependencies
+ */
+import {
+	normalizeShippingAddress,
+	normalizeOrderData,
+	getErrorMessageFromNotice,
+} from './utils';
+
+export const shippingAddressChangeHandler = async ( api, event ) => {
+	const response = await api.paymentRequestCalculateShippingOptions(
+		normalizeShippingAddress( event.shippingAddress )
+	);
+
+	// Possible statuses success, fail, invalid_payer_name, invalid_payer_email, invalid_payer_phone, invalid_shipping_address.
+	event.updateWith( {
+		status: response.result,
+		shippingOptions: response.shipping_options,
+		total: response.total,
+		displayItems: response.displayItems,
+	} );
+};
+
+export const shippingOptionChangeHandler = async ( api, event ) => {
+	const response = await api.paymentRequestUpdateShippingDetails( event );
```

Comment:
@ricardo it should've been `event.shippingOption`. I'm submitting a PR fixing it.